### DISCUSSION TITLE
Eclipse with raspi-maven archetype - adding delay before debugger attaches - app's not ready

### DISCUSSION THREAD

**Discussion by dannaughton** [Pi4J/pi4j#260] [General]
I have the raspi-maven archetype working in Eclipse - mostly.  But my debugger session tries to attach too quickly - before the application is running on the target.  (VM Connection refused).  If I launch a Remote Java Application debug session after the error - it works as expected.

Is there somewhere in the pom.xml or build.xml that I can add a pause?

Thanks all.

**Comment by taartspi** [comment]
Long time since I used Eclipse on my laptop and debugged the Pi target.
When you start the application on the pi include -agentlib:jdwp=transport=dt_socket,server=y,address=4000,server=y,suspend=y      The app will wait until your eclipse connects thru Pi port 4000       I can't find any notes on configuring the eclipse to use that specific port but it sounds like your question only pertains to the Pi app waiting.
